File: medplum-java/medplum-model/src/main/java/com/fhir/TestReport_Action1.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableTestReport_Action1.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface TestReport_Action1 {
  Optional<List<Extension>> extension();

  @JsonProperty("assert")
  Optional<TestReport_Assert> _assert();

  Optional<String> id();

  Optional<TestReport_Operation> operation();

  Optional<List<Extension>> modifierExtension();

  static ImmutableTestReport_Action1.Builder builder() {
    return ImmutableTestReport_Action1.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/TestReport_Action1.ts
import { Extension } from "./Extension";
import { TestReport_Assert } from "./TestReport_Assert";
import { TestReport_Operation } from "./TestReport_Operation";

export interface TestReport_Action1 {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    operation: TestReport_Operation | null | undefined;
    assert: TestReport_Assert | null | undefined;
}
